shared/query.go

```diff
@@ -45,11 +89,15 @@ func UserAgent() string {
 	return fmt.Sprintf("%s/%s", path.Base(info.Main.Path), info.Main.Version)
 }
 
+// QueryAPILimiter limits the rate that QueryAPI is called
+var QueryAPILimiter = rate.NewLimiter(rate.Every(time.Second)/5, 5)
+
 // QueryAPI is used to fetch data from the MangaDex API.
 func QueryAPI[T any](
 	ctx context.Context,
 	queryPath string,
 	queryParams url.Values,
+	limiter *rate.Limiter,
```

Comment:
Nothing ever seems to use this limiter field, so I think it should be removed. If you want to keep the functionality then move it to the Context instead.